src/gamma_inc.jl

```diff
@@ -38,61 +37,57 @@ const d80 = -.652623918595309E-03
    Uses the relation `gamma(a+1) = a*gamma(a)`.
 """
 function rgamma1pm1(a::Float64)
-    t=a
+    t = a
     rangereduce = a > 0.5
     t = rangereduce ? a-1 : a #-0.5<= t <= 0.5
     if t == 0.0
         return 0.0
     elseif t < 0.0
-        top = @horner(t , -.422784335098468E+00 , -.771330383816272E+00 , -.244757765222226E+00 , .118378989872749E+00 , .930357293360349E-03 , -.118290993445146E-01 , .223047661158249E-02 , .266505979058923E-03 , -.132674909766242E-03)
-        bot = @horner(t , 1.0 , .273076135303957E+00 , .559398236957378E-01)
+        top = @horner(t, -.422784335098468E+00, -.771330383816272E+00, -.244757765222226E+00, .118378989872749E+00, .930357293360349E-03, -.118290993445146E-01, .223047661158249E-02, .266505979058923E-03, -.132674909766242E-03)
+        bot = @horner(t, 1.0, .273076135303957E+00, .559398236957378E-01)
         w = top/bot
-        return rangereduce ? t*w/a : a*(w+1)
+        return rangereduce ? t*w/a : a*(w + 1)
     else
-        top = @horner(t , .577215664901533E+00 , -.409078193005776E+00 , -.230975380857675E+00 , .597275330452234E-01 , .766968181649490E-02 , -.514889771323592E-02 , .589597428611429E-03)
-        bot = @horner(t , 1.0 , .427569613095214E+00 , .158451672430138E+00 , .261132021441447E-01 , .423244297896961E-02)
+        top = @horner(t, .577215664901533E+00, -.409078193005776E+00, -.230975380857675E+00, .597275330452234E-01, .766968181649490E-02, -.514889771323592E-02, .589597428611429E-03)
+        bot = @horner(t, 1.0, .427569613095214E+00, .158451672430138E+00, .261132021441447E-01, .423244297896961E-02)
         w = top/bot
         return rangereduce ? (t/a)*(w - 1.0) : a*w
     end
 end
+
 """
     rgammax(a,x)
 
-Evaluation of ``1/\\Gamma(a) e^{-x} x^{a}``.
+Evaluation of ``1/\\Gamma(a) e^{-x} x^{a}``. Based on `DRCOMP` from the `NSWC` library.
 """
-function rgammax(a::Float64,x::Float64)
+function rgammax(a::Float64, x::Float64)
     if x == 0.0
         return 0.0
-    elseif a >= 20.0
-        u =x/a
-        if u == 0.0
+    elseif a > 20.0
+        t = x/a
+        if t == 0.0
             return 0.0
         end
-        t = (1.0/a)^2
-        t1 = (((0.75*t - 1.0)*t + 3.5)*t - 105.0)/(a*1260.0) #Using stirling Series : https://dlmf.nist.gov/5.11.1
-        t1 = t1 + a*LogExpFunctions.logmxp1(u)
-        if t1 >= exparg
-            return rt2pin*sqrt(a)*exp(t1)
-        end
+        w = -(stirling_error(a) - a*LogExpFunctions.logmxp1(t))
+        return 1/sqrt(2*Float64(π))*sqrt(a)*exp(w)
```

Comment:
I think the current version might be better because LLVM seems to be able to constant fold the `1/sqrt(2*Float64(π))` part
```julia
julia> foo(x) = 1/sqrt(2*Float64(π))*sqrt(x)
foo (generic function with 1 method)

julia> @code_llvm foo(2.0)
;  @ REPL[24]:1 within `foo'
define double @julia_foo_472(double %0) {
top:
; ┌ @ math.jl:582 within `sqrt'
; │┌ @ float.jl:371 within `<'
    %1 = fcmp uge double %0, 0.000000e+00
; │└
   br i1 %1, label %L7, label %L5

L5:                                               ; preds = %top
   %2 = call nonnull {}* @j_throw_complex_domainerror_474({}* inttoptr (i64 4775361824 to {}*), double %0)
   call void @llvm.trap()
   unreachable

L7:                                               ; preds = %top
; └
; ┌ @ math.jl:583 within `sqrt'
   %3 = call double @llvm.sqrt.f64(double %0)
; └
; ┌ @ float.jl:332 within `*'
   %4 = fmul double %3, 0x3FD9884533D43651
; └
  ret double %4
}
```